clang/test/SemaSYCL/attr-add-ir-attributes-function-conflict.cpp

```diff
@@ -0,0 +1,33 @@
+// RUN: %clang_cc1 -fsycl-is-device -fsyntax-only -verify %s
+
+// Tests that add_ir_attributes_function causes a warning when appearing with
+// potentially conflicting SYCL attributes.
+
+constexpr const char AttrName1[] = "Attr1";
+constexpr const char AttrVal1[] = "Val1";
+
+template <const char *... Strs> struct Wrapper {
+  template <typename KernelName, typename KernelType>
+  [[__sycl_detail__::add_ir_attributes_function(Strs...)]] __attribute__((sycl_kernel)) void kernel_single_task(const KernelType &kernelFunc) {
+    kernelFunc();
+  }
+};
```

Comment:
Does it make sense add a variant of `parallel_for`/`single_task` that allows to attach properties to kernel to mock headers?

Reply:
I worry that would be a bigger piece of work as we would also want to add a mock version of properties, then for completeness we also need to be able to handle properties on kernel functors through a `get` function. Maybe in the future if we need more complex testing in SemaSYCL for it, it might make sense, but for now I think it's overkill.

Reply:
OK.